integration/ksm/ksm_test.sh

```diff
@@ -0,0 +1,90 @@
+#!/bin/bash
+#
+# Copyright (c) 2019 Intel Corporation
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+set -e
+
+dir_path=$(dirname "$0")
+source "${dir_path}/../../lib/common.bash"
+source "${dir_path}/../../metrics/lib/common.bash"
+
+TEST_NAME="Run containers with ksm"
+KATA_KSM_THROTTLER="${KATA_KSM_THROTTLER:-yes}"
+RUNTIME="${RUNTIME:-kata-runtime}"
+PAYLOAD_ARGS="${PAYLOAD_ARGS:-tail -f /dev/null}"
+IMAGE="${IMAGE:-busybox}"
+WAIT_TIME="60"
+
+FIRST_TMP_FILE=$(mktemp firstcontainer.XXXX)
+SECOND_TMP_FILE=$(mktemp secondcontainer.XXXX)
+
+function remove_tmp_files() {
+	rm -rf $FIRST_TMP_FILE $SECOND_TMP_FILE
+}
+
+trap remove_tmp_files EXIT
+
+function turn_on_ksm() {
+	echo "Turning on KSM"
+	echo "1" > ${KSM_ENABLE_FILE}
+}
+
+function turn_off_ksm() {
+	echo "Turning off KSM"
+	echo "0" > ${KSM_ENABLE_FILE}
+}
+
+function setup() {
+	clean_env
+
+	check_processes
+
+	turn_on_ksm
+
+	check_for_ksm
+}
+
+function teardown() {
+	clean_env
+
+	check_processes
+
+	turn_off_ksm
+}
+
+function run_with_ksm() {
+	setup
+
+	# Running the first container
+	docker run -d --runtime=$RUNTIME $IMAGE $PAYLOAD_ARGS
+
+	echo "Entering KSM settle mode on first container"
+	wait_ksm_settle $WAIT_TIME > $FIRST_TMP_FILE
+
+	# Checking the pages merged for the first container
+	first_pages_merged=$(cat $FIRST_TMP_FILE | grep -w "check pages" | tail -1 | cut -d ' ' -f5)
```

Comment:
You could remove the `cat` here, but also you might want to add `|| true` if this could ever fail?:

```sh
first_pages_merged=$(grep -w "check pages" "$FIRST_TMP_FILE" | tail -1 | cut -d ' ' -f5 || true)
```

Reply:
I'd probably forego the whole tmpfile/cat/grep thing, and just directly read from `/sys/kernel/mm/ksm/pages_shared`. That gets rid of the tmpfiles completely and we don't have to go parsing logs....
Whilst there, I'd probably add that `pages_shared` file to the block of KSM vars at https://github.com/kata-containers/tests/blob/master/metrics/lib/common.bash#L20-L23, and also then use that in the metrics `common.bash` file as well (where we already hardwire the path - which looks like an oversight on my part...)